### ISSUE TITLE
Document new 1.2 liveness/readiness probe options

### ISSUE THREAD

**Issue by therc** [kubernetes/kubernetes#22193]
See #15967

**Comment by timstclair** [comment]
What sort of documentation are you looking for? The parameters are already documented in the [API reference](http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_probe), and these are advanced features which I'm not sure belong in the user guide examples.

**Comment by marcolenzo** [comment]
@timstclair 

Could we document at least the criteria used to consider probes successful?

I am using `httpGet` probes and I am working under the assumption that kubernetes is just looking at the response code and expecting 200? Is this correct?

I think a separate issue could be giving the option of configuring expectation similar to LTM monitors.

**Comment by talgendler** [comment]
Can you please add a reference somewhere [here](http://kubernetes.io/docs/user-guide/liveness/)
I know that I wanted to configure periodic timeout  so I started to look for it and eventually found what I wanted in this thread (by looking through several issues), however if there were a link to API it would be helpful. I also believe that not everyone expects more options are available.

**Comment by timstclair** [comment]
Sure. It looks like all the information is there, but spread across multiple documents:

- http://kubernetes.io/docs/user-guide/pod-states/#container-probes
- http://kubernetes.io/docs/user-guide/production-pods/#liveness-and-readiness-probes-aka-health-checks
- http://kubernetes.io/docs/user-guide/liveness/
- http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_probe

/cc @devin-donnelly

**Comment by talgendler** [comment]
@timstclair well actually none of the guides mentions a way to configure how often to perform the probe. I think that if every page you mentioned above had a link to the relevant API it would help a lot. This is how I got to this thread:

1. First I found this page: (https://github.com/kubernetes/kubernetes/issues/12866)
2. From there I got to (https://github.com/kubernetes/kubernetes/pull/15967) 
3. And finally to this thread.
